src/core/reducers/abuse.js

```diff
@@ -0,0 +1,95 @@
+/* @flow */
+export const LOAD_ADDON_ABUSE_REPORT = 'LOAD_ADDON_ABUSE_REPORT';
+export const SEND_ADDON_ABUSE_REPORT = 'SEND_ADDON_ABUSE_REPORT';
+
+type LoadAddonAbuseReportType = {
+  addon: {|
+    guid: string,
+    id: number,
+    slug: string,
+  |},
+  message: string,
+  reporter: Object | null,
+};
+
+export function loadAddonAbuseReport(
+  { addon, message, reporter }: LoadAddonAbuseReportType
+) {
+  if (!addon) {
+    throw new Error('addon is required');
+  }
+  if (!message) {
+    throw new Error('message is required');
+  }
+  if (reporter === undefined) {
+    throw new Error('reporter cannot be undefined');
+  }
+
+  return {
+    type: LOAD_ADDON_ABUSE_REPORT,
+    payload: { addon, message, reporter },
+  };
+}
+
+type SendAddonAbuseReportAction = {|
+  addonSlug: string,
+  errorHandlerId: string,
+  message: string,
+|};
+
+export function sendAddonAbuseReport(
+  { addonSlug, errorHandlerId, message }: SendAddonAbuseReportAction
+) {
+  if (!addonSlug) {
+    throw new Error('addonSlug is required');
+  }
+  if (!errorHandlerId) {
+    throw new Error('errorHandlerId is required');
+  }
+  if (!message) {
+    throw new Error('message is required');
+  }
+
+  return {
+    type: SEND_ADDON_ABUSE_REPORT,
+    payload: { addonSlug, errorHandlerId, message },
+  };
+}
+
+export const initialState = {
+  bySlug: {},
+};
+
+type ReducerState = {|
+  bySlug: {
+    [addonSlug: string]: Array<{| message: string, reporter: string |}>,
```

Comment:
The way you have it now is not an Array, it's

```js
[addonSlug: string]: {| message: string, reporter: string |},
```

Reply:
Aha, right. I copied your flow thing and wasn't paying attention. My bad.